be/src/object_store/s3_uri.h

```diff
@@ -0,0 +1,73 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+#pragma once
+
+#include <string>
+#include <string_view>
+
+namespace starrocks {
+
+// Creates a new S3URI in the form of scheme://bucket/key?query#fragment
+// The URI supports any valid URI schemes to be backwards compatible with s3a and s3n,
+// and also allows users to use S3ObjectStore with other S3-compatible object storage services.
+class S3URI {
+public:
+    explicit S3URI(std::string location) : _location(std::move(location)) {}
+
+    bool parse();
+
+    // REQUIRE: has been `parse()`d
+    std::string_view schema() const { return _schema; }
+
+    // REQUIRE: has been `parse()`d
+    std::string_view bucket() const { return _bucket; }
+
+    // REQUIRE: has been `parse()`d
+    std::string_view object() const { return _object; }
```

Comment:
Seems `key` is more reasonable.
https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/s3/AmazonS3URI.html

Reply:
updated